File: 백준/Silver/12537. Closing the Loop （Small）/Closing the Loop （Small）.py
import sys
input = sys.stdin.readline

def main():
    N = int(input())
    for i in range(N):
        T = int(input())
        rope_list = input().split()
        red_list, blue_list = [], []
        for rope in rope_list:
            rope_len = int(rope[:-1])
            if rope[-1] == 'R':
                red_list.append(rope_len)
            else:
                blue_list.append(rope_len)
                
        red_list.sort(reverse=True)
        blue_list.sort(reverse=True)
        
        total = 0
        for j in range(min(len(red_list), len(blue_list))):
            total += red_list[j] + blue_list[j]
            total -= 2
            
        if total < 0:
            total = 0
            
        print(f"Case #{i+1}: {total}")
    
    
if __name__ == "__main__":
    main()

File: 백준/Silver/12537. Closing the Loop （Small）/Closing the Loop （Small）.java
import java.io.*;
import java.util.*;
import java.lang.Math;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer tokenizer;

        Integer T = Integer.parseInt(reader.readLine());

        Integer N;
        for (int i = 0; i < T; ++i) {
            N = Integer.parseInt(reader.readLine());
            tokenizer = new StringTokenizer(reader.readLine());

            String token, len, color;
            ArrayList<Integer> red_list = new ArrayList<>();
            ArrayList<Integer> blue_list = new ArrayList<>();
            for (int j = 0; j < N; ++j) {
                token = tokenizer.nextToken();
                len = token.substring(0, token.length() - 1);
                color = token.substring((token.length() - 1));

                if (color.equals("R")) red_list.add(Integer.parseInt((len)));
                else blue_list.add(Integer.parseInt(len));
            }

            Collections.sort(blue_list, Collections.reverseOrder());
            Collections.sort(red_list, Collections.reverseOrder());

            Integer total = 0;
            Integer size = Math.min(red_list.size(), blue_list.size());
            for (int k = 0; k < size; ++k) {
                total += blue_list.get(k) + red_list.get(k);
                total -= 2;
            }

            if (total < 0) total = 0;

            System.out.printf("Case #%d: %d\n", i + 1, total);

        }
    }
}